src/integer_mod_q/z_q/fmpz_mod_helpers.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains helpful functions on [`fmpz`] values in a ring/`modulus` context.
+
+use crate::integer::fmpz_helpers::distance;
+use flint_sys::fmpz::{fmpz, fmpz_cmp};
+
+const ZERO_FMPZ: fmpz = fmpz(0);
+
+/// Computes the shortest distance of `self` to the next zero instance
+/// regarding the `modulus`.
+///
+/// WARNING: This function assumes `value` to be reduced,
+/// i.e. `0 <= value <= modulus`.
```

Comment:
a) The reduced interval is shorter.
```suggestion
/// WARNING: This function assumes `value` to be reduced,
/// i.e. `0 <= value < modulus`.
```

b) The actual allowed interval is larger then specified one (ca. -modulus/2 < value < 1.5 * modulus) since you are using distance instead of  a subtraction
=> I would suggest to use a subtraction since that is faster.

Reply:
a) Yes, the reduced interval is shorter, but if the value == modulus, the function handles this just fine.
b) I do not understand your point. I use subtraction for the distance.